src/components/panes/__tests__/side-pane.test.js

```diff
@@ -0,0 +1,100 @@
+import { shallow } from 'enzyme'
+import React from 'react'
+
+import Drawer from 'material-ui/Drawer'
+import { MuiThemeProvider } from 'material-ui/styles'
+import Resizable from 're-resizable'
+
+import { SidePaneUnconnected, mapStateToProps } from '../side-pane'
+import tasks from '../../../actions/task-definitions'
+
+describe('SidePaneUnconnected React component', () => {
+  let props
+  let mountedPane
+
+  const sidePane = () => {
+    if (!mountedPane) {
+      mountedPane = shallow(<SidePaneUnconnected {...props} />)
+    }
+    return mountedPane
+  }
+
+  beforeEach(() => {
+    props = {
+      sidePaneMode: '_APP_INFO',
+      title: 'App info',
+      sidePaneWidth: 562,
+      openOnMode: '_APP_INFO',
+      task: tasks.toggleAppInfoPane,
+    }
+    mountedPane = undefined
+  })
+
+  it('always renders one MuiThemeProvider', () => {
+    expect(sidePane().find(MuiThemeProvider).length).toBe(1)
+  })
+
+  it('always renders one Drawer inside MuiThemeProvider', () => {
+    expect(sidePane().wrap(sidePane().find(MuiThemeProvider))
+      .find(Drawer)).toHaveLength(1)
+  })
+
+  it("sets the Drawer's open prop to be true for correct props", () => {
+    props.sidePaneMode = '_APP_INFO'
+    props.openOnMode = '_APP_INFO'
+    expect(sidePane().find(Drawer).props().open)
+      .toBe(true)
+  })
+
+  it("sets the Drawer's open prop to be false for incorrect props", () => {
+    props.sidePaneMode = '_APP_INFO'
+    props.openOnMode = 'history'
+    expect(sidePane().find(Drawer).props().open)
+      .toBe(false)
+  })
+
+  it('always renders one Resizable inside Drawer', () => {
+    expect(sidePane().wrap(sidePane().find(Drawer))
+      .find(Resizable)).toHaveLength(1)
+  })
+
+  it("sets the Resizable's onResizeStop prop to correct function", () => {
+    // const changeSidePaneWidth = jest.spyOn(tasks, 'changeSidePaneWidth.callback')
```

Comment:
What should be the correct approach to test https://github.com/iodide-project/iodide/blob/f5605943d226c040354ee244f7429b0e7c0cbfeb/src/components/panes/side-pane.jsx#L62-L63

Reply:
not sure tbh. @bcolloran?